frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditImage/UploadImageInput.tsx

```diff
@@ -0,0 +1,69 @@
+import { useEffect } from 'react'
+import { ControllerRenderProps } from 'react-hook-form'
+import { forwardRef } from '@chakra-ui/react'
+import imageCompression from 'browser-image-compression'
+
+import {
+  MAX_UPLOAD_FILE_SIZE,
+  MB,
+  VALID_UPLOAD_FILE_TYPES,
+} from '~shared/constants/file'
+
+import Attachment from '~components/Field/Attachment'
+
+import { EditImageInputs } from './EditImage'
+
+type UploadImageInputProps = Pick<
+  ControllerRenderProps<EditImageInputs, 'attachment'>,
+  'name' | 'onChange' | 'value'
+> & {
+  onError?: (errMsg: string) => void
+}
+
+export const UploadImageInput = forwardRef<UploadImageInputProps, 'div'>(
+  ({ name, onChange, value, onError }, ref): JSX.Element => {
+    useEffect(() => {
+      return () => {
+        if (!value?.srcUrl) return
+        URL.revokeObjectURL(value.srcUrl)
+      }
+    }, [value?.srcUrl])
+
+    const handleChange = (file?: File) => {
+      if (!file) return onChange({ file, srcUrl: undefined })
+
+      if (file.name.split('.').pop() === 'gif') {
```

Comment:
Small simplification, and also account for potential weirdness in case

```suggestion
      if (/\.gif$/i.test(file.name)) {
```

Reply:
4631733